analyzers/tests/SonarAnalyzer.UnitTest/TestCases/InvocationResolvesToOverrideWithParams.cs

```diff
@@ -2,161 +2,408 @@
-        {
-            MyMethod(""); // Compliant
-        }
+// See https://github.com/SonarSource/sonar-dotnet/issues/2234
+public class FuncAndActionCases
+{
+    static void Main(string[] args)
+    {
+        M1(() => Console.WriteLine("hi"));
     }
 
-    public class Test2
+    public static void M1(params Action[] a) { }
+    public static void M1<T>(Func<T> f) { }
+    public static void M1(Func<Task> f) { }
+}
+
+public class WithLocalFunctions
+{
+    public static void Test(int foo, params object[] p)
     {
-        public static implicit operator Test2(string s) { return null; }
+        Console.WriteLine("test1");
+    }
 
-        public void MyMethod(params string[] s) { }
-        public void MyMethod(Test2 s) { }
+    public static void Test(double foo, object p1)
+    {
+        Console.WriteLine("test2");
+    }
 
-        public Test2()
+    public void Method()
+    {
+        static void Call()
         {
-            MyMethod(""); // Noncompliant
+            Test(42, null); // Noncompliant {{Review this call, which partially matches an overload without 'params'. The partial match is 'void WithLocalFunctions.Test(double foo, object p1)'.}}
         }
     }
+}
 
-    // See https://github.com/SonarSource/sonar-dotnet/issues/2234
-    public class FuncAndActionCases
+// Reproducer for https://github.com/SonarSource/sonar-dotnet/issues/5430
+namespace Repro5430
+{
+    public class SomeClass
     {
-        static void Main(string[] args)
+        private int InlineInitializedField11 = new SomeClass("1", "s1").PrivateOverload("s1");                  // Compliant
+        private int InlineInitializedField12 = new SomeClass("1", "s1").PrivateOverload("s1", "s2");            // Noncompliant
+        private int InlineInitializedField13 = new SomeClass("1", "s1").PrivateOverload(null, "s2");            // Noncompliant
+        private int InlineInitializedField14 = new SomeClass("1", "s1").PrivateOverload(null, new[] { "s2" });  // Compliant
+        private int InlineInitializedField15 = new SomeClass("1", "s1").PrivateOverload("42", "s1", "s2");      // Compliant
+
+        private int InlineInitializedField21 = new SomeClass("1", "s1").InternalOverload("s1");                 // Compliant
+        private int InlineInitializedField22 = new SomeClass("1", "s1").InternalOverload("s1", "s2");           // Noncompliant
+        private int InlineInitializedField23 = new SomeClass("1", "s1").InternalOverload(null, "s2");           // Noncompliant
+        private int InlineInitializedField24 = new SomeClass("1", "s1").InternalOverload(null, new[] { "s2" }); // Compliant
+        private int InlineInitializedField25 = new SomeClass("1", "s1").InternalOverload("42", "s1", "s2");     // Compliant
+
+        private SomeClass(object a, string b) { }
+        private SomeClass(string a, string b) { }
+        public SomeClass(string a, params string[] bs) { }
+
+        private int PrivateOverload(object a, string b) => 1492;
```

Comment:
To remove the question "what effect has that specific value on the behavior?"
```suggestion
        private int PrivateOverload(object a, string b) => 42;
```
or
```suggestion
        private int PrivateOverload(object a, string b) => 0;
```